lambdas/common-layer/nodejs/node_modules/dev-portal-common/customers-controller.js

```diff
@@ -1040,13 +1040,17 @@ const deleteApiKeysByIdentityId = async identityId => {
   )
 }
 
-const findUserInUserPool = async userId => {
-  return cognitoIdp
-    .adminGetUser({
-      Username: userId,
+const getEmailForUserSub = async userSub => {
+  const found = await cognitoIdp
+    .listUsers({
+      Filter: `sub = "${userSub}"`,
+      Limit: 1,
```

Comment:
Why?

Reply:
Couple reasons:

1. I have a `sub`, not the username itself, and I'd rather future-proof it. In fact, while creating this, I found that using subs as usernames wouldn't always find the right user - they'd commonly either not find a user or (worse) get the *wrong* user.
2. I need the email, and this gives me more direct access than AdminGetUser. I don't have to iterate a list to get there, and I found `findUserInPool` and `__getCognitoUserEmail` ended up much simpler.